src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudService.kt

```diff
@@ -52,10 +53,30 @@ class CosmasGoogleCloudService(private val bucketName: String,
         responseObserver.onCompleted()
     }
 
+    override fun createPatch(request: CosmasProto.CreatePatchRequest,
+                             responseObserver: StreamObserver<CosmasProto.CreatePatchResponse>) {
+        LOG.info("Get request for create new patch of file # ${request.fileId} by user ${request.userId}")
+        synchronized(this.fileBuffer) {
+            val project = this.fileBuffer.getValue(request.projectId)
+            val newPatch = CosmasProto.FileInformation.Patch.newBuilder().setText(request.text)
+                    .setTimeStamp(request.timeStamp).setUserId(request.userId).build()
+            val fileInformation = project[request.fileId]
+            if (fileInformation != null) {
+                project[request.fileId] = CosmasProto.FileInformation.newBuilder()
```

Comment:
I think that `fileInformation.toBuilder()` is more canonical and efficient way of making a mutable copy.

Reply:
ok